### PR TITLE
Add support for parallel tests

### PR DESCRIPTION
**Pull request by malcolmbaig**
## Context
For faster test runs, we want to be able to run our test suite in parallel using multiple cores.

## Changes proposed in this pull request

- Add the [parallel_tests](https://github.com/grosser/parallel_tests) gem
- Update config for Postgres, Capybara, and Redis
- Document how to run the test suite in parallel locally

## Guidance to review
- Per-commit gives a good breakdown of the changes
- Does the doc make sense?
- Run it locally! Are you seeing any issues?

## Link to Trello card
https://trello.com/c/ySrLogAF

## Things to check

- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training#azure-hosting-devops-pipeline)

### CODE REVIEW

File: docs/parallel_tests.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
+# Parallel Tests
+
+This document explains how to run the test suite across multiple cores in development.
+
+## Prerequisites
+
+### Check environment
+
+Ensure that REDIS_URL is not present in .env.test or .env.
```
Comment (davidgisbey) [inline docs/parallel_tests.md:9]: Why .env?

Comment (davidgisbey) [inline docs/parallel_tests.md:9]: why in .env?

**Review by davidgisbey — APPROVED** [review]
Just set it up and ran the tests locally. They were significantly quicker. Like wayyyy quicker. Thanks for all your work on this Malcolm!

I only have 8 cores btw so if someone has 16 it might be worth them following that section of the doc.

**Review by duncanjbrown — APPROVED** [review]
🏎️ This is great!  `1m46s` for the whole suite on my machine. And docs too 👌👌👌.

I liked having the same output in and out of `parallel_tests` so I copied the `--format` line in `.rspec` to `.rspec_parallel` — would you consider adopting that?

(Only 8 cores here too).

**Comment by malcolmbaig** [comment]
@davidgisbey @duncanjbrown Excellent, thank you both for testing.

> I only have 8 cores btw so if someone has 16 it might be worth them following that section of the doc.

Had to do this locally, can confirm the fix works nicely.

> I liked having the same output in and out of parallel_tests so I copied the --format line in .rspec to .rspec_parallel — would you consider adopting that?

I haven't quite figured out how to collate test failures from each process and print them all at the end. With `progress` formatting, there's much less scrolling back to identify what's failed. Happy to change over to `documentation` if we can find a nicer way to output errors.

**Pull request merged**